ui/src/app/utils/index.tsx

```diff
@@ -630,3 +630,21 @@ export const useId = () => {
   const [id] = useState(() => fp.uniqueId('element-'));
   return id;
 };
+
+const nothing = {};
+
+// maybe - takes a function and a value. If the value is not defined returns "nothing"
+// When this was written is was not used outside this file, so it was not exported
+// Feel free to export this when a proper use case arises
+// Example usage: fp.flow(maybe(doSomethingIfIhaveData), maybe(doAnotherThingIfThereIsAResult))(getData())
+const maybe = fp.curry((fn, value) => value !== nothing && value ? fn(value) : nothing);
```

Comment:
I am surprised that this is not already part of Typescript.  Thanks for the add, this should be useful.